File: apps/api-crm/Models/Ticket.cs
namespace Crm.Api.Models;

public class Ticket
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    public string Status { get; set; } = "Unclaimed";
    public Guid CustomerId { get; set; }
    public string? AssignedToId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public CustomerProfile Customer { get; set; } = null!;
    public User? AssignedTo { get; set; }
    public ICollection<Message> Messages { get; set; } = [];
}

File: apps/web-crm/src/types/ticket.ts
export type TicketStatus = "Unclaimed" | "Claimed" | "Ongoing" | "Completed" | "Canceled";

export interface TicketListItem {
  id: string;
  title: string;
  status: TicketStatus | string;
  customerName: string;
  unreadMessageCount: number;
  createdAt: string;
}

export interface Ticket {
  id: string;
  title: string;
  description: string;
  imageUrl?: string | null;
  status: TicketStatus | string;
  customerId: string;
  customerName: string;
  assignedToId?: string | null;
  assignedToName?: string | null;
  createdAt: string;
  updatedAt: string;
}
